src/compiler/transformers/es6.ts

```diff
@@ -549,7 +549,7 @@ namespace ts {
                     /*modifiers*/ undefined,
                     createVariableDeclarationList([
                         createVariableDeclaration(
-                            getDeclarationName(node),
+                            node.name || getGeneratedNameForNode(node),

```

Comment:
It shouldn't other than comments. Note that I've reverted back to calling `getDeclarationName`, but changed the behavior of `getDeclarationName`.